test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,363 @@
+        j2.create_eatcpu_job(30)
+        jid2 = self.server.submit(j2)
+
+        self.server.expect(JOB, {'job_state': 'R'}, jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        self.server.delete(jid2, wait=True)
+        self.server.expect(JOB, {'job_state': 'F'},
+                           offset=40, extend='x', id=jid1)
+
+        j3 = Job(TEST_USER, a)
+        j3.create_eatcpu_job()
+        jid3 = self.server.submit(j3)
+        self.server.expect(JOB, {'job_state': 'R'}, jid3)
+        self.server.delete(jid3, wait=True)
+
+        # No R record ; Only E record for job1 which finished
+        self.server.accounting_match(
+            msg='E;' + jid1 +
+            '.*Exit_status=0.*resources_used.*run_count=1', id=jid1,
+            regexp=True)
+        self.server.accounting_match(
+            msg='R;' + jid1, id=jid1, existence=False,
+            max_attempts=10)
+
+        # No R record, No E record for job2 which is in 'Q' state
+        self.server.accounting_match(
+            msg='R;' + jid2, id=jid2, existence=False,
+            max_attempts=10)
+        self.server.accounting_match(
+            msg='E;' + jid2, id=jid2, existence=False,
+            max_attempts=10)
+
+        # No R record ; Only E record for job3 which was deleted
+        # when in 'R' state
+        self.server.accounting_match(
+            msg='R;' + jid3, id=jid3, existence=False,
+            max_attempts=10)
+        self.server.accounting_match(
+            msg='E;' + jid3 +
+            '.*Exit_status=271.*resources_used.*run_count=1', id=jid3,
+            regexp=True)
+
+    def test_acclog_mom_down(self):
+        """
+        Check accounting logs when node is down and MoM is restarted
+        """
+        a = {ATTR_nodefailrq: 15}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        # Submit a job
+        a = {'Resource_List.select': '1:ncpus=1:mem=20gb'}
+        j = Job(TEST_USER, a)
+        j.create_eatcpu_job()
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, jid)
+
+        # Submit a job array
+        ja = Job(TEST_USER, attrs={
+            ATTR_J: '1-2',
+            'Resource_List.select': 'ncpus=1:mem=20gb'}
+        )
+        ja.create_eatcpu_job()
+        jid_a = self.server.submit(ja)
+
+        subjid1 = j.create_subjob_id(jid_a, 1)
+        subjid2 = j.create_subjob_id(jid_a, 2)
+
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
+
+        self.assertTrue(self.server.isUp())
+        self.assertTrue(self.mom.isUp())
+
+        # kill -9 mom
+        self.mom.signal('-KILL')
+
+        # Verify that nodes are reported to be down.
+        self.server.expect(NODE, {ATTR_NODE_state: 'down'},
+                           id=self.mom.shortname, offset=15)
```

Comment:
Now with TPP, the server should notice the mom go down immediately.

Reply:
Removed offset